app/src/main/res/layout/message_threads_layout.xml

```diff
@@ -23,34 +23,52 @@
       <LinearLayout
         android:id="@+id/mailbox_and_unread_layout"
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:layout_centerVertical="true"
+        android:layout_height="wrap_content"
         android:animateLayoutChanges="true"
         android:clipToPadding="false"
         android:focusable="true"
         android:importantForAccessibility="yes"
-        android:orientation="vertical"
-        android:paddingBottom="@dimen/grid_9"
-        android:paddingEnd="@dimen/grid_2"
-        android:paddingStart="@dimen/grid_3"
-        android:paddingTop="@dimen/grid_9"
+        android:nextFocusDown="@id/mailbox_switch"
+        android:orientation="horizontal"
+        android:paddingEnd="@dimen/activity_horizontal_margin"
+        android:paddingStart="@dimen/activity_horizontal_margin"
+        android:layout_marginTop="@dimen/grid_9"
+        android:layout_marginBottom="@dimen/activity_vertical_margin"
         app:layout_collapseMode="parallax"
         app:layout_collapseParallaxMultiplier="0.2">
 
-        <TextView
-          android:id="@+id/mailbox_text_view"
-          style="@style/Title1"
-          android:layout_width="wrap_content"
+        <LinearLayout
+          android:layout_width="0dp"
           android:layout_height="wrap_content"
-          tools:text="Inbox" />
+          android:layout_weight="1"
+          android:orientation="vertical">
+
+          <TextView
+            android:id="@+id/mailbox_text_view"
+            style="@style/Title1"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            tools:text="Inbox" />
+
+          <TextView
+            android:id="@+id/unread_count_text_view"
+            style="@style/FootnotePrimaryMedium"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textColor="@color/accent"
+            tools:text="3 unread" />
+
+        </LinearLayout>
 
-        <TextView
-          android:id="@+id/unread_count_text_view"
-          style="@style/FootnotePrimaryMedium"
+        <Button
+          android:id="@+id/mailbox_switch"
```

Comment:
I think this could be `switch_inbox_button` or `switch_mailbox_button` since it's a button and not a switch